src/promptflow/promptflow/_sdk/_service/app.py

```diff
@@ -167,8 +146,49 @@ def monitor_request():
                         kill_exist_service(port)
                     break
 
-        initialize_created_by_info()
+        # Retrieve created_by info and cache it in advance to avoid blocking the first request.
+        get_created_by_info_with_cache()
         if not sys.executable.endswith("pfcli.exe"):
             monitor_thread = ThreadWithContextVars(target=monitor_request, daemon=True)
             monitor_thread.start()
     return app, api
+
+
+created_by_for_local_to_cloud_trace = {}
+created_by_for_local_to_cloud_trace_lock = threading.Lock()
+
+
+def get_created_by_info_with_cache():
+    if len(created_by_for_local_to_cloud_trace) > 0:
+        return created_by_for_local_to_cloud_trace
+    trace_provider = Configuration.get_instance().get_trace_provider()
+    if trace_provider is None or extract_workspace_triad_from_trace_provider(trace_provider) is None:
```

Comment:
If exception raised from extract_workspace_triad_from_trace_provider, should we catch it or just let it failed?

Reply:
I think function here should consider this, for example, I have launched PFS, and later I updated the pf config with wrong value; in this case, PFS cannot know this, so here need to handle this?

Reply:
In this case, could we add context info rightly?
We just try to get created_by info after getting cosmosDB token.

Reply:
We decide skip checking about Configuration.
Just try to get created_by info as trying to write CosmosDB.